app/controllers/posts_controller.rb

```diff
@@ -18,7 +18,8 @@ def create
           redirect_to course_posts_path(params[:course_id])
         else
           flash[:warning] = "Post couldn't be created" 
-          render 'new'
+          redirect_to(new_course_post_path(params[:id]), alert: 'Post couldnt be created') and return
```

Comment:
This looks good. Just so I have a better understanding. Why did you make this change?